executor/adapter.go

```diff
@@ -568,6 +569,32 @@ func (a *ExecStmt) handlePessimisticDML(ctx context.Context, e Executor) error {
 	}
 }
 
+// GetTimestampWithRetry tries to get timestamp using retry and backoff mechanism
+func (a *ExecStmt) GetTimestampWithRetry(ctx context.Context) (uint64, error) {
+	tsoMaxBackoff := 15000
+	bo := tikv.NewBackoffer(context.Background(), tsoMaxBackoff)
```

Comment:
Would you like to move line after 580 and use the passed ctx argument instead of `context.Background()` ?